cmd/create.go

```diff
@@ -286,3 +288,41 @@ func filterYaml(cmd *cobra.Command, args []string, toComplete string) ([]string,
 	}
 	return []string{"yaml", "yml"}, cobra.ShellCompDirectiveFilterFileExt
 }
+
+// Determines if overwrite is allowed
+func checkOverwriteAllowed(depDir string, bp config.Blueprint, overwriteFlag bool) error {
+	if _, err := os.Stat(depDir); os.IsNotExist(err) {
+		return nil // all good, no previous deployment
+	}
+
+	useWErr := fmt.Errorf("deployment folder %q already exists, use -w to overwrite", depDir)
+	// try to get previous deployment
+	expPath := filepath.Join(modulewriter.ArtifactsDir(depDir), modulewriter.ExpandedBlueprintName)
+	prev, _, err := config.NewDeploymentConfig(expPath)
+	if err != nil { // malformed previous deployment, rely on flag
+		if overwriteFlag {
+			return nil
+		}
+		return useWErr
+	}
+
+	newGroups := map[config.GroupName]bool{}
+	for _, g := range bp.DeploymentGroups {
+		newGroups[g.Name] = true
+	}
+
+	for _, g := range prev.Config.DeploymentGroups {
+		if !newGroups[g.Name] {
+			return fmt.Errorf("you are attempting to remove a deployment group %q, which is not supported", g.Name)
+		}
+	}
+
+	if prev.Config.GhpcVersion != bp.GhpcVersion {
+		logging.Info("WARNING: ghpc_version has changed from %q to %q", prev.Config.GhpcVersion, bp.GhpcVersion)
+	}
+
+	if !overwriteFlag {
```

Comment:
Without thinking too hard about it I would put these checks in the following order:
1. warn on ghpc version - as just a warning no harm to always check
2. check for overwrite flag - if they have not approved overwrite then maybe we should not give warnings about deleting groups
3. error on group removal

Is there rational to use the current order?